test/harness.ts

```diff
@@ -48,6 +48,12 @@ for (let key in bunEnv) {
 
 delete bunEnv.NODE_ENV;
 
+if (isDebug) {
+  // This makes debug build memory leak tests more reliable.
+  // The code for dumping out the debug build transpiled source code has leaks.
+  bunEnv.BUN_DEBUG_NO_DUMP = "1";
```

Comment:
i think we should have this but can we also fix the memory leak in the source code dump.

line 222: add `defer bun.default_allocator.free(source_file);`